modules/openideal_user_registration/openideal_user_registration.module

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * @file
+ * Contains openideal_idea.module.
+ */
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\user\UserInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function openideal_user_registration_form_user_login_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $form['actions']['social_buttons'] = [
+    '#theme' => 'login_with',
+    '#social_networks' => \Drupal::config('social_auth.settings')->get('auth'),
+    '#weight' => '1',
+  ];
+}
+
+/**
+ * Implements hook_user_login().
+ */
+function openideal_user_registration_faq_user_login(UserInterface $account) {
+  if (strpos(\Drupal::request()->getRequestUri(), 'user/registrationpassword')) {
```

Comment:
First of all, it's always better to check by route name instead of page URL.
Secondly, I think it would be better to detect the first login by `$user->getLastLoginTime()`. This is how they do that in Drupal core.